api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiPatientFetcher.java

```diff
@@ -9,8 +10,16 @@
 
     /**
      * Perform query to mpi server to get patient by identifier & convert it to local patient.
+     * @param patientIdentifier patient identifier
+     * @return converted patient not yet saved to local DB.
+     */
+    Patient fetchMpiPatient(PatientIdentifier patientIdentifier);
+
+    /**
+     * Perform query to mpi server to get patient by specific identifier & convert it to local patient.
      * @param patientId patient identifier
+     * @param identifierTypeUuid uuid of {@link org.openmrs.PatientIdentifierType }
      * @return converted patient not yet saved to local DB.
      */
-    Patient fetchMpiPatient(String patientId);
+    Patient fetchMpiPatient(String patientId, String identifierTypeUuid);
```

Comment:
I thought MPIs use the other fields in to get an identifier domain and not uuids

Reply:
they can also use uuids. This is for the case that you already know that there is a specific patient that you want to get and you already know that this patient is in the MPI. This is used after registration as the MPI global domain identifier is not sent back in the PIX response, thus if we want to save the MPI global domain identifier (although this is not necessarily the case as we are discussing above) then we need to send another pdq to get it and we do that through this message. Another case where this is used is when importing patients